src/components/checkboxGroupCollapsible/index.tsx

```diff
@@ -0,0 +1,89 @@
+import React, { Fragment } from 'react';
+
+import { Box, Collapse, IconButton, useDisclosure } from '@chakra-ui/react';
+
+import Stack from '../stack';
+import { ChevronDownLargeIcon } from '../icons';
+import Divider from '../divider';
+import { Item, State } from '../checkboxFilter/type';
+import CheckboxWithOptions from '../checkboxFilter/CheckboxWithOptions';
+
+interface CheckboxCollapsibleProps<ItemKey extends string> {
+  parent: Item<ItemKey>;
```

Comment:
since you go with parent/child naming. Can we do something like `parentCheckbox` / `childCheckboxes`?